### PR TITLE
#158628186 Add test cases for challenge 3

### PR DESCRIPTION
**Pull request by meshack-mbuvi**
#### What does this PR do?
Creates tests for various edge cases for the api endpoints
#### Description of Task to be completed?
Have tests for the following endpoints `failing`
```
- POST /auth/signup  Register a user   
- POST /auth/login  Login a user 
- GET /rides   Fetch all available rides   
- GET /rides/<rideId>  Fetch the details of a single ride  
- POST /rides/<rideId>/requests  Make a ride request  This endpoint helps a user to indicate interest in a specific available ride. 
- POST /users/rides 
- GET /users/rides/<rideId>/requests  Fetch all ride requests
- PUT /users/rides/<rideId>/requests/<requestId>  Accept or reject a ride request. 
```
#### How should this be manually tested?
using your terminal,
- clone this repo
- `cd ride-my-way`
- set and activate your virtual environment
- `git checkout ft-challenge-3-tests-158628186`
- Run `nosetests`

#### Any background context you want to provide?
Unit testing gives peace of mind as most of the bugs are identified and corrected at early stages of product development.
#### What are the relevant pivotal tracker stories?
[[#158628186]](https://www.pivotaltracker.com/story/show/158628186)

#### How to Review
Look at the files named `testUserAccount.py` and `testRides.py`. Give your feedback.

### CODE REVIEW

**Comment by Ogollah** [comment]
This is fine.

File: tests/testUserAccount.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            "username": "musyoka",
+            "driver": True,
+            "password": "mbuvi1",
+            "phone": "0719800509",
+            "confirm password": "mbuvi1"
+        }
+        self.passwords = {
+            "email": "meshmbuvi@gmail.com",
+            "username": "mbuvi1",
+            "driver": False,
```
Comment (JoshuaOndieki) [inline tests/testUserAccount.py]: The variable namings between line 25 - 56 doesn't sound right.
Like the `self.passwords` variable I thought I will see passwords only to realize it means there's some change in the password but the dictionary maintains user data as its content.

Comment (meshack-mbuvi) [reply tests/testUserAccount.py]: Thanks for the correction.Its updated now

**Comment by xcixor** [comment]
Nice follow up of tdd

**Comment by rethes** [comment]
Your code is okay

File: tests/testUserAccount.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        }
+        self.data_with_unmatching_passwords = {
+            "email": "meshmbuvi@gmail.com",
+            "username": "mbuvi1",
+            "driver": False,
+            "password": "mbuvi1",
+            "phone": "0719800509",
+            "confirm password": "mbuvi11"
+        }
+        self.data_with_invali_email = {
```
Comment (xcixor) [inline tests/testUserAccount.py]: typo 'invali'

File: tests/testUserAccount.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
+import json
+import unittest
+from psycopg2 import connect
+
+from application import create_app, db as database
+from application.models import dbname, user, host, password
+
+
+class SignTests(unittest.TestCase):
```
Comment (xcixor) [inline tests/testUserAccount.py]: you can also test for password length

Comment (meshack-mbuvi) [reply tests/testUserAccount.py]: done on user sign-up feature

**Comment by LWanjiru** [comment]
Why are the Travis builds failing?

**Comment by meshack-mbuvi** [comment]
These are tests without the functionalities. AM working on the functionalities
